### PR TITLE
avoid dinput reinitialization on unrelated WM_DEVICECHANGE events

### PR DESCRIPTION
**Pull request by fr500**
## Guidelines

1. Rebase before opening a pull request
2. If you are sending several unrelated fixes or features, use a branch and a separate pull request for each
3. If possible try squashing everything in a single commit. This is particularly beneficial in the case of feature merges since it allows easy bisecting when a problem arises

## Description

avoid dinput reinitialization on unrelated WM_DEVICECHANGE events

## Related Issues

I have faced this issue for years making RA on windows unusable on one of my PCs that has lots of network connections.

Basically RetroArch would reinitialize the whole input subsistem on WM_DEVICECHANGE without considering the actual event. Now we check if a device node is actually changed before triggering a reinit so hotplug functionality should remain intact.

## Reviewers

@bparker06 @twinaphex

### CODE REVIEW

**Comment by twinaphex** [comment]
Nice improvement.
